src/base/memory_monitor_sr.hpp

```diff
@@ -40,12 +47,18 @@ namespace cubmem
       memory_monitor &operator = (const memory_monitor &) = delete;
       memory_monitor &operator = (memory_monitor &&) = delete;
 
+    public:
+      size_t get_alloc_size (char *ptr);
+
     private:
       static std::string make_tag_name (const char *file, const int line);
 
     private:
       std::string m_server_name;
+      int magic;
```

Comment:
How about using char*? I'm suggesting to make it const.
```suggestion
      static constexpr char* magic = "MMON";
      static constexpr int magic_size = 4;
```

Reply:
I think magic string makes a lot of calls `strcmp()` for checking about the memory is in the memory_monitor scope. So I decided to make it integer.

Reply:
Fair enough